docs/old-reference-guide/modules/messaging-concepts/pages/timeouts.adoc

```diff
@@ -0,0 +1,233 @@
+
+Both warnings and timeouts are logged at the `WARN` level.
+The message will contain:
+
+. The name of the handler with the message, or component of transaction
+. The time it has been processing so far
+. The time it has left before the timeout is reached
+. The stack trace of the handler or transaction, starting from where the timeout started
+
+For example, a warning message could look like this:
+
+[source,text]
+----
+2025-02-12T20:01:20.795Z  WARN 68040 --- [playground-correlation] [ axon-janitor-0] axon-janitor                             : Message [io.axoniq.playground.publisher.MyContinuousEvent] for handler [io.axoniq.playground.publisher.PersistentStreamEventProcessor] is taking long time to process. Current time: [5000ms]. Will be interrupted in [5000ms].
+Stacktrace of current thread:
+java.base/java.lang.Thread.sleep0(Native Method)
+java.base/java.lang.Thread.sleep(Thread.java:509)
+io.axoniq.playground.publisher.PersistentStreamEventProcessor.handle(PersistentStreamEventProcessor.kt:16)
+java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
+/** Removed some part of the stack trace for brevity **/
+org.axonframework.messaging.timeout.TimeoutWrappedMessageHandlingMember.handle(TimeoutWrappedMessageHandlingMember.java:61)
+----
+
+This is beneficial for debugging and monitoring purposes, as it allows you to see which handlers are taking a long time to process.
+Additionally, it helps prevent a broken process from blocking processing of other messages.
+Examples of this are http libraries awaiting a response, that never comes during an event handler.
+
+=== Defaults and disabling
+
+As timeouts and warnings have been introduced in a minor version (4.11), the default values have been set to a high value to not interfere with any processes that users have already built.
+
+Long-running handlers or transactions will start logging warnings after 10 seconds, and will be interrupted after 30 seconds.
+You can always change these settings to your liking, or disable the warnings completely as below.
+Note that this also disables the annotation-based timeouts.
+
+[tabs]
+======
+Spring Boot::
++
+--
+For Spring Boot, disabling all timeouts and warnings can be done by setting the following property in your `application.properties` or `application.yml`:
+
+[source,properties]
+----
+axon.timeout.enabled=false
+----
+
+--
+
+
+Axon Configuration::
++
+--
+To disable timeouts and warnings for non-Spring Boot applications, you only have to not register the configuration outlined on this page, as it's not activated by default.
+--
+======
+
+[#handler-timeouts]
+== Handler timeouts
+
+The method you define for Axon Framework to invoke is known as a message handler.
+These are the methods you annotate with `@CommandHandler`, `@EventHandler` or `@QueryHandler`.
+
+As of Axon Framework 4.11.0, you can set a timeout on all message handlers, depending on the message type.
```

Comment:
I would omit the "4.11.0"-part since the documentation is already versioned.
```suggestion
You can set a timeout on all message handlers, with unique configuration per message type.
```